src/sentry/snuba/metrics/datasource.py

```diff
@@ -440,19 +462,28 @@ def get_single_metric_info(
 
 
 def get_tags(
-    projects: Sequence[Project], metric_names: Optional[Sequence[str]], use_case_id: UseCaseKey
+    projects: Sequence[Project], metrics: Optional[Sequence[str]], use_case_id: UseCaseKey
 ) -> Sequence[Tag]:
     """Get all metric tags for the given projects and metric_names"""
     assert projects
 
     try:
-        tags, _ = _fetch_tags_or_values_per_ids(
-            projects=projects,
-            metric_names=metric_names,
-            column="tags.key",
-            referrer="snuba.metrics.meta.get_tags",
-            use_case_id=use_case_id,
-        )
+        if len(metrics) and all(MRI_SCHEMA_REGEX.match(metric) for metric in metrics):
```

Comment:
Can you add a test that validates the code path for `all` MRI being valid?

Reply:
Just to make sure we have no regressions if people mess up the code, it's quite an important validation for us to do, otherwise the whole system will fail downstream.